### PR TITLE
AM-34284 HDH: Generate, Issue, Charge Invoice

### PR DESCRIPTION
**Pull request by simformsolutions**
PR contains changes for
Account creation, Invoice Generate, Issue, Charge under v1 api

### CODE REVIEW

File: lib/Paid.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
 'use strict';
 
 var _ = require('lodash');
+var Path = require("path");
+var config = require(Path.resolve('./', "config"))
```
Comment (ktemo) [inline lib/Paid.js]: missing semicolon

File: lib/base/APIMethod.js [hunk 2]
```diff
@@ -79,9 +79,9 @@ APIMethod.prototype = _.extend(APIMethod.prototype, {
     return Promise.defer();
   },
 
-  execute: function() {
+  execute: function(asJson) {
     var deferred = this.createDeferred();
-    Request(this.method, this.url(), this.params, this.headers, this.onSuccess(deferred), this.onError(deferred));
+    Request(this.method, this.url(), this.params, this.headers, this.onSuccess(deferred), this.onError(deferred),asJson);
```
Comment (ktemo) [inline lib/base/APIMethod.js]: missing space

File: lib/base/Request.js [hunk 3]
```diff
@@ -3,18 +3,23 @@
...
-    opts.qs = params;
-  } else {
-    opts.form = params;
+  if(_.isUndefined(asJson)){
+    if(method == "GET" || method == "DELETE") {
+      opts.qs = params;
+    } else {
+      opts.form = params;
+    }
+  }else{
```
Comment (ktemo) [inline lib/base/Request.js]: missing spaces

File: lib/resources/Invoice.js [hunk 4]
```diff
@@ -23,18 +23,18 @@ var staticApi = {
...
     var methodArgs = APIResource.prepareArguments(arguments, Invoice.create);
-    return new APIMethod("post", "/invoices", methodArgs, undefined, APIResource.constructBldr(_.bind(Invoice, null, this.apiKey)), this.apiKey).execute();
+    return new APIMethod("post", "/invoices", methodArgs, undefined, APIResource.constructBldr(_.bind(Invoice, null, this.apiKey)), this.apiKey).execute(true);
   },
 
   // These are roughly the same as the ones on the prototype, but don't require a new Invoice().
   issue: function(id, params, headers) {
     var methodArgs = APIResource.prepareArguments(arguments, Invoice.issue);
-    return new APIMethod("post", "/invoices/:id/issue", methodArgs, undefined, APIResource.constructBldr(_.bind(Invoice, null, this.apiKey)), this.apiKey).execute();
+    return new APIMethod("patch", "/invoices/:id", methodArgs, undefined, APIResource.constructBldr(_.bind(Invoice, null, this.apiKey)), this.apiKey).execute();
```
Comment (ktemo) [inline lib/resources/Invoice.js]: is this ok? we are changing the endpoint to use?

Comment (simformsolutions) [reply lib/resources/Invoice.js]: Yes, in v1 endpoints, issue=true, charge=true will be passed as methodArgs instead endpoint

Comment (simformsolutions) [reply lib/resources/Invoice.js]: However i am rolling back those changes as we will use invoice.update method for issue=true or charge=true

**Pull request merged**